src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/vector/elements.hpp

```diff
@@ -0,0 +1,274 @@
+// Copyright 2009-present MongoDB, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#ifndef _WIN32
+#include <sys/param.h> // Endian detection
+#endif
+
+#include <cstdint>
+#include <cstring>
+
+#include <bsoncxx/config/prelude.hpp>
+
+namespace bsoncxx {
+namespace v_noabi {
+namespace vector {
+
+namespace iterators {
+
+template <typename Iterator>
+class packed_bit_element;
+
+template <typename Iterator>
+class packed_bit_byte;
+
+} // namespace iterators
+
+namespace elements {
+
+/// @brief A 32-bit float value in packed little-endian format
+class float32 {
+   public:
+    /// @brief Construct a packed little-endian value from a float input in the local byte order.
+    /// @param value Floating point value to convert
+    float32(float value) noexcept {
+#if defined(_WIN32) ||                                                                                              \
+    (defined(__BYTE_ORDER__) && defined(__ORDER_LITTLE_ENDIAN__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)) || \
+    defined(__BYTE_ORDER) && defined(__LITTLE_ENDIAN) && (__BYTE_ORDER == __LITTLE_ENDIAN)
+        std::memcpy(bytes, &value, sizeof value);
+#elif (defined(__BYTE_ORDER__) && defined(__ORDER_BIG_ENDIAN__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)) || \
+    defined(__BYTE_ORDER) && defined(__BIG_ENDIAN) && (__BYTE_ORDER == __BIG_ENDIAN)
```

Comment:
We should probably get around to defining our own [std::endian](https://en.cppreference.com/w/cpp/types/endian) at some point. 😅